twingate/internal/client/service-account.go

```diff
@@ -72,6 +72,19 @@ func (client *Client) UpdateServiceAccount(ctx context.Context, serviceAccount *
 }
 
 func (client *Client) DeleteServiceAccount(ctx context.Context, serviceAccountID string) error {
+	serviceAccount, err := client.ReadServiceAccount(ctx, serviceAccountID)
+	if err != nil && !errors.Is(err, ErrGraphqlResultIsEmpty) {
+		return err
+	}
+
+	if serviceAccount != nil {
+		for _, key := range serviceAccount.Keys {
+			if err := client.RevokeServiceKey(ctx, key); err != nil {
+				return err
+			}
+		}
+	}
+
 	opr := resourceServiceAccount.delete()
```

Comment:
So before this PR, we ignores the SA's keys and 
- Immediately deleted it from TF state
- Called `DeleteServiceAccount` mutation - which would have returned an error, because you cannot delete an SA with valid keys.

So it seems to me the problem here that TF state gets updated regardless to the result of the delete mutation?

Reply:
Seems like an issue with all the delete methods?  (`DeleteGroup`, `DeleteResource` etc seem to behave the same)